File: Programmers/Lv0/다음에올숫자.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> common) {
    if (common[0] + common[2] == 2 * common[1]) //등차
        return common.back() + (common[1] - common[0]);
    else // 등비 common[0] * common[2] == common[1] * commom[1]
        return common.back() * (common[1] / common[0]);
}

File: Programmers/Lv0/다음에올숫자.py
def solution(common):
    if (common[0] + common[2] == 2 * common[1]):
        return common[len(common) - 1] + (common[1] - common[0])
    else:
        return common[len(common) - 1] * (common[1] / common[0])